### PR TITLE
[openimageio] Fix find and use dependencies

### LINKED ISSUE #19878: openimageio cmake errors
When building against the latest version of openimageio in vcpkg there is an error 
```
CMake Error at /Users/jmacey/vcpkg/installed/x64-osx/share/OpenImageIO/OpenImageIOConfig.cmake:33 (elseif):
  given arguments:

    "2.5.0" "VERSION_GREATER_EQUAL" "2.4" "AND"
```
It seems the line

```elseif (2.5.0 VERSION_GREATER_EQUAL 2.4 AND )```

is the issue (I presume the dangling AND) this is an autocreated file so not sure where the error stems from.

**Environment**

Mac, Windows and Linux (RHEL 7.7 tested)

**To Reproduce**

Create a dummy project and add the following to CMakeLists.txt
```cmake
find_package(OpenImageIO CONFIG REQUIRED)
target_link_libraries(test PRIVATE OpenImageIO::OpenImageIO )
```
run cmake

**Expected behavior**
It shoudl build

### LINKED ISSUE #19884: [openimageio] PNG::PNG not being found.
**Describe the bug**

After removing the spurious `AND` from this bug [#19878](https://github.com/microsoft/vcpkg/issues/19878) and building a simple demo I get the following errors 
```
CMake Error at /Users/jmacey/vcpkg/scripts/buildsystems/vcpkg.cmake:536 (_add_executable):
  Target "Image" links to target "PNG::PNG" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:16 (add_executable)
```
libpng and ecm are installed via vcpkg and if I remove `\$<LINK_ONLY:PNG::PNG>;` from the OpenImageIOTargets.cmake file and manually link lib png by using 
```cmake
find_package(libpng CONFIG REQUIRED)
target_link_libraries(Image PRIVATE png_static)
```
Everything builds links and runs fine. This didn't happen on previous versions (I only update every 6 months or so).
 

**Environment**
Mac, Windows and Linux

**To Reproduce**

./vcpkg install openimageio

Try building a demo agains it (I have attached a minimal demo).

**Expected behavior**
Should build
[CMakeLists.txt](https://github.com/microsoft/vcpkg/files/7093445/CMakeLists.txt)
[main.txt](https://github.com/microsoft/vcpkg/files/7093453/main.txt)

### PR DESCRIPTION
**Pull request by JackBoosY**
1. Fix find dependency `libpng` and `libheif`.
2. Restore `FindOpenEXR.cmake` since it calls `find_dependency(OpenEXR CONFIG)` and it contains the required macro `FOUND_OPENEXR_WITH_CONFIG` in `OpenImageIOConfig.cmake`.
3. Add dependency port `libsquish`.
4. Add 3 `find_dependency` in `OpenImageIOConfig.cmake` to fix usage issues.
5. Install `FindLibsquish.cmake` since libsquish and cmake doesn't provide the find method.
6. Fix `vcpkg-cmake-wrapper` install path since [we always use it which in the lower-case port name path](https://github.com/microsoft/vcpkg/blob/master/scripts/buildsystems/vcpkg.cmake#L724-L727) and to use `FindLibsquish.cmake` in `OpenImageIO` directory.

Fixes #19878 #19884.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@jmacey Can you please test my changes?

Thanks.

**Comment by jmacey** [comment]
Testing now but this branch still has the glib https://github.com/microsoft/vcpkg/issues/19558  error which I need to fix first.

On 2 Sep 2021, at 07:59, Jack·Boos·Yu ***@***.******@***.***>> wrote:



@jmacey<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjmacey&data=04%7C01%7Cjmacey%40bournemouth.ac.uk%7C9684a00de3404f80cb1108d96ddf2ab9%7Cede29655d09742e4bbb5f38d427fbfb8%7C0%7C0%7C637661627527533942%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=PptJOkhRUb598z2jkXlyMX1mJvq0XmJyeb2BDGQPnqc%3D&reserved=0> Can you please test my changes?

Thanks.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fvcpkg%2Fpull%2F19916%23issuecomment-911284374&data=04%7C01%7Cjmacey%40bournemouth.ac.uk%7C9684a00de3404f80cb1108d96ddf2ab9%7Cede29655d09742e4bbb5f38d427fbfb8%7C0%7C0%7C637661627527543942%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=7iYuYj7IgnSSP6ewfAc28CuFXF%2BfTWOGMKRPT4a0nMI%3D&reserved=0>, or unsubscribe<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAACK4G5WZXYIPCSE3DXF4TLT74OD3ANCNFSM5DILDGDQ&data=04%7C01%7Cjmacey%40bournemouth.ac.uk%7C9684a00de3404f80cb1108d96ddf2ab9%7Cede29655d09742e4bbb5f38d427fbfb8%7C0%7C0%7C637661627527543942%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=r6lSQxLVr37M%2FvjYA54VvamjfxBdtskx3u%2FOuw5hSgw%3D&reserved=0>.
Triage notifications on the go with GitHub Mobile for iOS<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fapps.apple.com%2Fapp%2Fapple-store%2Fid1477376905%3Fct%3Dnotification-email%26mt%3D8%26pt%3D524675&data=04%7C01%7Cjmacey%40bournemouth.ac.uk%7C9684a00de3404f80cb1108d96ddf2ab9%7Cede29655d09742e4bbb5f38d427fbfb8%7C0%7C0%7C637661627527553937%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=LWVdMblAerXcgv2Cw6Sv2cRkQzLN5pB9ZJGy9fvcAno%3D&reserved=0> or Android<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fplay.google.com%2Fstore%2Fapps%2Fdetails%3Fid%3Dcom.github.android%26referrer%3Dutm_campaign%253Dnotification-email%2526utm_medium%253Demail%2526utm_source%253Dgithub&data=04%7C01%7Cjmacey%40bournemouth.ac.uk%7C9684a00de3404f80cb1108d96ddf2ab9%7Cede29655d09742e4bbb5f38d427fbfb8%7C0%7C0%7C637661627527553937%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=MSHc2QGsORrPpzDrrlGPPYDGVmTf1CCVq9iXuwx3yJw%3D&reserved=0>.



________________________________

BU is a Disability Confident Employer and has signed up to the Mindful Employer charter. Information about the accessibility of University buildings can be found on the BU AccessAble webpages.
This email is intended only for the person to whom it is addressed and may contain confidential information. If you have received this email in error, please notify the sender and delete this email, which must not be copied, distributed or disclosed to any other person.
Any views or opinions presented are solely those of the author and do not necessarily represent those of Bournemouth University or its subsidiary companies. Nor can any contract be formed on behalf of the University or its subsidiary companies via email.

**Comment by jmacey** [comment]
Am now getting errors about libSquish. 

Could not find a package configuration file provided by "Libsquish" with
  any of the following names:

    LibsquishConfig.cmake
    libsquish-config.cmake

Neither of these files are in the vcpg subdirs however libsquish is installed. 

```
./vcpkg install libsquish
Computing installation plan...
The following packages are already installed:
    libsquish[core]:x64-linux -> 1.15-7
Package libsquish:x64-linux is already installed

Total elapsed time: 14.46 us

```

but note that is gives no cmake info

```
./vcpkg install openimageio
Computing installation plan...
The following packages are already installed:
    openimageio[core]:x64-linux -> 2.2.17.0#2
Package openimageio:x64-linux is already installed

Total elapsed time: 14.39 us

The package openimageio:x64-linux provides CMake targets:

    find_package(Modules CONFIG REQUIRED)
    target_link_libraries(main PRIVATE OpenImageIO::OpenImageIO)

    find_package(OpenImageIO CONFIG REQUIRED)
    target_link_libraries(main PRIVATE OpenImageIO::OpenImageIO OpenImageIO::OpenImageIO_Util)

    find_package(openimageio CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Libsquish::Libsquish)

```

File: ports/openimageio/portfile.cmake [hunk 1]
```diff
@@ -83,6 +83,7 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/doc"
                     "${CURRENT_PACKAGES_DIR}/debug/share")
 
 file(COPY "${SOURCE_PATH}/src/cmake/modules/FindOpenImageIO.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
+file(COPY "${SOURCE_PATH}/src/cmake/modules/FindLibsquish.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
```
Comment (JackBoosY) [inline ports/openimageio/portfile.cmake:87]:
```suggestion
file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/openimageio")
```

**Comment by JackBoosY** [comment]
@jmacey  This PR will not fix issue https://github.com/microsoft/vcpkg/issues/19558 because I don't find out the reason.
And it should be good now for the Libsquish issue.

File: versions/o-/openimageio.json [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "9f421bd53327f1d197a1175e98a343300d604733",
```
Comment (JackBoosY) [inline versions/o-/openimageio.json:4]:
```suggestion
      "git-tree": "5e2d1ed44e9b26fc1283eb9a3ca14c2564d095e3",
```

File: versions/o-/openimageio.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "be0e222ad1e0394f30553dfb112a04d7e6a5e1aa",
```
Comment (JackBoosY) [inline versions/o-/openimageio.json:4]:
```suggestion
      "git-tree": "c627b98b5edc66758b3ea5a128292fd2f2a41393",
```

**Review by PhoebeHui — APPROVED** [review]

File: ports/openimageio/portfile.cmake [hunk 4]
```diff
@@ -83,7 +83,8 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/doc"
                     "${CURRENT_PACKAGES_DIR}/debug/share")
 
 file(COPY "${SOURCE_PATH}/src/cmake/modules/FindOpenImageIO.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
-file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
+file(COPY "${SOURCE_PATH}/src/cmake/modules/FindLibsquish.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```
Comment (PhoebeHui) [inline ports/openimageio/portfile.cmake:86]:
```suggestion
file(COPY "${SOURCE_PATH}/src/cmake/modules/FindLibsquish.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
```

File: ports/openimageio/portfile.cmake [hunk 5]
```diff
@@ -83,7 +83,8 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/doc"
...
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/openimageio")
```
Comment (JackBoosY) [inline ports/openimageio/portfile.cmake:87]:
```suggestion
file(COPY "${SOURCE_PATH}/src/cmake/modules/FindLibsquish.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/OpenImageIO")
file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
```

File: versions/o-/openimageio.json [hunk 6]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "c627b98b5edc66758b3ea5a128292fd2f2a41393",
```
Comment (JackBoosY) [inline versions/o-/openimageio.json:4]:
```suggestion
      "git-tree": "cf0cdb6b6753dbd307add667ffab26719506b65a",
```

**Comment by JackBoosY** [comment]
Tested this PR success:
```cmake
cmake_minimum_required(VERSION 3.10)

project(sample)
find_package(OpenImageIO CONFIG REQUIRED)
add_executable(sample test.cpp)
target_link_libraries(sample PRIVATE OpenImageIO::OpenImageIO OpenImageIO::OpenImageIO_Util)
```
result:
```
root@usr:/home/usr/work/test# ./../vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/bin/cmake CMakeLists.txt  -DCMAKE_TOOLCHAIN_FILE=/home/usr/work/19916/vcpkg/scripts/buildsystems/vcpkg.cmake
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
CMake Warning (dev) at /home/usr/work/19916/vcpkg/installed/x64-linux/share/zlib/vcpkg-cmake-wrapper.cmake:13 (_find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  CMake variable ZLIB_ROOT is set to:

    /home/usr/work/19916/vcpkg/installed/x64-linux

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  /home/usr/work/19916/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
  /home/usr/work/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /home/usr/work/19916/vcpkg/installed/x64-linux/share/openexr/OpenEXRConfig.cmake:29 (find_dependency)
  /home/usr/work/19916/vcpkg/installed/x64-linux/share/openexr/vcpkg-cmake-wrapper.cmake:1 (_find_package)
  /home/usr/work/19916/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
  /home/usr/work/vcpkg/downloads/tools/cmake-3.20.2-linux/cmake-3.20.2-linux-x86_64/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  /home/usr/work/19916/vcpkg/installed/x64-linux/share/OpenImageIO/OpenImageIOConfig.cmake:36 (find_dependency)
  /home/usr/work/19916/vcpkg/installed/x64-linux/share/openimageio/vcpkg-cmake-wrapper.cmake:4 (_find_package)
  /home/usr/work/19916/vcpkg/scripts/buildsystems/vcpkg.cmake:738 (include)
  CMakeLists.txt:5 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found ZLIB: /home/usr/work/19916/vcpkg/installed/x64-linux/lib/libz.a (found version "1.2.11")
-- Found Libsquish: /home/usr/work/19916/vcpkg/installed/x64-linux/include
-- Found ZLIB: /home/usr/work/19916/vcpkg/installed/x64-linux/lib/libz.a (found suitable version "1.2.11", minimum required is "1.2.11")
-- Configuring done
-- Generating done
-- Build files have been written to: /home/usr/work/test
root@usr:/home/usr/work/test# make
[ 50%] Building CXX object CMakeFiles/sample.dir/test.cpp.o
[100%] Linking CXX executable sample
[100%] Built target sample
```

**Comment by BillyONeal** [comment]
Thanks for the fix!

**Comment by jmacey** [comment]
I've just done a git pull on the master and removed the old downloads and installs and still get this error

CMake Error at C:/Users/jpmac/vcpkg/installed/x64-windows/share/OpenImageIO/OpenImageIOConfig.cmake:33 (elseif):
  given arguments:

    "2.5.0" "VERSION_GREATER_EQUAL" "2.4" "AND"

If I edit the OpenImageIOConfig.cmake and add

elseif (2.5.0 VERSION_GREATER_EQUAL 2.4 AND FOUND_OPENEXR_WITH_CONFIG )

It all works fine, so it seems that there is still some sort of error with this fix in the master. 

.\vcpkg.exe --version
Vcpkg package management program version 2021-08-12-85ab112d5ee102bc6eac8cdbbfdd173a71374e04

**Comment by JackBoosY** [comment]
That's caused by wrongly merge to master.

**Pull request merged**